api/tests/e2e/e2e_suite_test.go

```diff
@@ -159,15 +161,12 @@ func createOrgRaw(orgName, authHeader string) *http.Response {
 }
 
 func createOrg(orgName, authHeader string) presenter.OrgResponse {
-	resp := createOrgRaw(orgName, authHeader)
-	Expect(resp).To(HaveHTTPStatus(http.StatusCreated))
-	Expect(resp).To(HaveHTTPHeaderWithValue(headers.ContentType, "application/json"))
-	defer resp.Body.Close()
-
 	org := presenter.OrgResponse{}
-	err := json.NewDecoder(resp.Body).Decode(&org)
-	Expect(err).NotTo(HaveOccurred())
-
+	api.Request(http.MethodPost, "/v3/organizations").
```

Comment:
How about 
```
api.CreateOrgRequest()
.WithBody(orgPayload(orgName)).
....
```

We would thus not have to remember the verb and the endpoint of an operation which might be annoying

Reply:
I kind of like the explicitness of it, because it makes it very obvious what the test is doing. If we define a `CreateXXXRequest` for each restful resource the helper will end up having a million methods, but we could use the constants from the handlers and have something like 
```
api.CreateOrgRequest()
api.PostRequest(apis.OrgsEndpoint).
...
```
if you thing this is a better direction...

Reply:
My point was that with the explicit API, even using constants, you would have to remember whether it is a post/put/patch, whatever. If we do not have endpoints serving multiple HTTP verbs we could have something like

```
api.Request(apis.CreateOrgEndpoint).....
```

and let the `Request` method figure out whether it should be a post or put or patch (via maybe supporting an internal map between endpoint and HTTP verb)

Reply:
Yep, I get that. It is true that we have to remember the http verb, but as I said I kind of like this explicitness - that you can tell that this test tests `POST "/v3/organizations"` which maps nicely to what is written in the story acceptance.

The lack of explicitness is what I do not like about our current tests. As a matter of fact our current set of functions already offer the ease of use that you are after:
```
org = createOrg(orgName, tokenAuthHeader)
``` 

but they lack explicitness and sometimes when the argument list is too long can even be confusing:
```
role := createOrgRole("organization_manager", rbacv1.UserKind, userName, org.GUID, tokenAuthHeader)
```
vs
```
api.Request(http.MethodPost, "/v3/roles").
    WithBody(userOrgRolePayload("organization_user", userName, org.GUID)).
    DoWithAuth(tokenAuthHeader).
    ValidateStatus(http.StatusCreated)
```